npm/pkg/dataplane/ipsets/ipset.go

```diff
@@ -372,21 +372,49 @@ func (set *IPSet) hasMember(memberName string) bool {
 	return isMember
 }
 
-func (set *IPSet) getSetIntersection(existingIntersection map[string]struct{}) (map[string]struct{}, error) {
-	if !set.canSetBeSelectorIPSet() {
-		return nil, npmerrors.Errorf(
-			npmerrors.IPSetIntersection,
-			false,
-			fmt.Sprintf("[IPSet] Selector IPSet cannot be of type %s", set.Type.String()))
+// affiliatedIPs returns the IPs for a hash set or the IPs of the member sets for a list set
+// This method, intersectAffiliatedIPs, and GetSetContents are good examples of how the
+// ipset struct may have been better designed as an interface with hash and list implementations.
+// Not worth it to redesign though.
+func (set *IPSet) affiliatedIPs() map[string]struct{} {
+	if set.Kind == HashSet {
+		result := make(map[string]struct{}, len(set.IPPodKey))
+		for ip := range set.IPPodKey {
+			result[ip] = struct{}{}
```

Comment:
can we use the set.IPPodKey map directly ignore values instead of creating a map with the same keys?

Reply:
this has to be a separate map because we'll be removing members from it based on IPPodKeys of other sets later

Reply:
thinking about this more, there are 2 scenarios:
1. when determining which IPs to apply a new policy on, we use the full map
2. when updating the IPSets that a pod is a part of, we only check to see if the Pod IP is in the map

The second scenario doesn't require the full map and could use O(1) space, but I'm not sure if this a worthwhile optimization if we already do the O(N) space option for another scenario